core/src/apps/ethereum/sign_tx.py

```diff
@@ -22,8 +25,31 @@
 MAX_CHAIN_ID = 2147483629
 
 
+def format_access_list(access_list):
+    mapped = map(
+        lambda item: [
+            address.bytes_from_address(item.address),
+            list(map(lambda s: unhexlify(s[2:]), item.storage_keys)),
+        ],
+        access_list,
+    )
+    return list(mapped)
+
+
 @with_keychain_from_chain_id
 async def sign_tx(ctx, msg, keychain):
+
+    if msg.tx_type == 2:
+        return await sign_eip1559_tx(ctx, msg, keychain)
+
+    # Remove empty access list for legacy tx
+    if len(msg.access_list) == 0:
```

Comment:
what happens if access_list is not empty for legacy tx?

Reply:
This is sort of an edge case with `protobuf`. Because if you pass `access_list = None` in the `EthereumSignTx` message, it turns it into an empty array. But for legacy transactions we expect the `access_list = None`, if not, the RLP length is wrong and thereby also the signature.

Reply:
Reading this again, I probably should just set `access_list = None` before entering legacy signing anyway. In case the user has passed something the signature will be invalid.

Reply:
I have simplified the logic in `get_total_length` to consider this instead, which imo is cleaner. Thanks for making me aware!

Reply:
> But for legacy transactions we expect the `access_list = None`, if not, the RLP length is wrong and thereby also the signature.

no idea where you got this: `sign_legacy_tx` doesn't touch the `access_list` field so it should not care about whether there's an empty list or anything else

Reply:
> I have simplified the logic in `get_total_length` to consider this instead, which imo is cleaner. Thanks for making me aware!

oh, i see now! yes, that looks cleaner :)